### PR TITLE
4.12 multicore configs

### PR DESCRIPTION
**Pull request by ctk21**
This PR adds configurations to install `4.12.0+domains+effects` and `4.12.0+domains`
It also includes this new naming which we hope will be easier for people to understand.

### CODE REVIEW

File: packages/ocaml-variants/ocaml-variants.4.12.0+domains/opam [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+]
+conflict-class: "ocaml-core-compiler"
+flags: compiler
+setenv: CAML_LD_LIBRARY_PATH = "%{lib}%/stublibs"
+build: [
+  ["./configure" "--prefix" "%{prefix}%" "--enable-debug-runtime" "--disable-warn-error"]
+  ["%{make}%" "-j%{jobs}%" "world"]
+  ["%{make}%" "-j%{jobs}%" "world.opt"]
+]
+install: ["%{make}%" "install"]
```
Comment (dra27) [inline packages/ocaml-variants/ocaml-variants.4.12.0+domains/opam:18]:
```suggestion
build: [
  ["./configure" "--prefix=%{prefix}%" "--disable-warn-error"]
  [make "-j%{jobs}%"]
]
install: [make "install"]
```
(also modernising!)

File: packages/ocaml-variants/ocaml-variants.4.12.0+domains+effects/opam [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+]
+conflict-class: "ocaml-core-compiler"
+flags: compiler
+setenv: CAML_LD_LIBRARY_PATH = "%{lib}%/stublibs"
+build: [
+  ["./configure" "--prefix" "%{prefix}%" "--enable-debug-runtime" "--disable-warn-error"]
+  ["%{make}%" "-j%{jobs}%" "world"]
+  ["%{make}%" "-j%{jobs}%" "world.opt"]
+]
+install: ["%{make}%" "install"]
```
Comment (dra27) [inline packages/ocaml-variants/ocaml-variants.4.12.0+domains+effects/opam:18]:
```suggestion
build: [
  ["./configure" "--prefix=%{prefix}%" "--disable-warn-error"]
  [make "-j%{jobs}%"]
]
install: [make "install"]
```
(modernising!)

**Review by dra27 — COMMENTED** [review]
`"ocaml-options-vanilla" {post}` should prevent installing `ocaml-option-flambda` and friends. The rest is just some updating w.r.t. upstream opam-repository.

**Comment by ctk21** [comment]
@dra27 thanks for the review, I've applied your suggestions.

**Comment by Sudha247** [comment]
I'm afraid `ocaml-lsp-server` and `merlin` don't work with `4.12+*` because of this dependency: `ocaml < 4.12`. This appears to be the case with ocaml-lsp on stock OCaml as well, but merlin works with stock OCaml 4.12.0.

So for now, do we just mention that melin and ocaml-lsp work only with 4.10.0+multicore and open an issue to track this?

**Comment by ctk21** [comment]
Added @dra27's option magic which allows people to try variants (e.g. afl or flambda).

**Pull request merged**